src/read/macho/file.rs

```diff
@@ -68,14 +69,15 @@ where
                         sections.push(MachOSectionInternal::parse(index, section));
                     }
                 } else if let Some(symtab) = command.symtab()? {
-                    symbols = symtab.symbols(endian, data)?;
+                    symbols = symtab.symbols(endian, linkedit_data)?;
```

Comment:
It might be time to stop trying to share code between `parse` and `parse_at`, and instead change `parse_at` to be something that is specifically parsing an image inside a dyld shared cache. So it would be passed in a reference to the `DyldCacheImage` (or something like that), and this code would know how to find the correct data for the load command (so no need to preparse the load commands with `linkedit_addr`).

Reply:
That is an excellent idea.